test/integration/provider/rti.go

```diff
@@ -0,0 +1,139 @@
+package provider
+
+import (
+	"fmt"
+
+	gcp "github.com/gardener/machine-controller-manager-provider-gcp/pkg/gcp"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ResourcesTrackerImpl struct {
+	InitialVolumes   []string
+	InitialInstances []string
+	InitialMachines  []string
+	MachineClass     *v1alpha1.MachineClass
+	SecretData       map[string][]byte
+	ClusterName      string
+}
+
+func (r *ResourcesTrackerImpl) InitializeResourcesTracker(machineClass *v1alpha1.MachineClass, secretData map[string][]byte, clusterName string) error {
+
+	clusterTag := "kubernetes-io-cluster-" + clusterName
+	clusterTagValue := "1"
+	r.MachineClass = machineClass
+	r.SecretData = secretData
+	r.ClusterName = clusterName
+
+	ms := gcp.NewGCPPlugin(&gcp.PluginSPIImpl{})
+	ctx, svc, err := ms.SPI.NewComputeService(&corev1.Secret{Data: secretData})
+	if err != nil {
+		return err
+	}
+
+	instances, err := GetInstancesWithTag(ctx, svc, "mcm-integration-test-true", machineClass, secretData)
+	if err == nil {
+		r.InitialInstances = instances
+		volumes, err := GetAvailableVolumes(ctx, svc, clusterTag, clusterTagValue, machineClass, secretData)
+		if err == nil {
+			r.InitialVolumes = volumes
+			r.InitialMachines, err = GetMachines(machineClass, secretData)
+			return err
+		} else {
+			return err
+		}
+	} else {
+		return err
+	}
```

Comment:
Else block is not needed here if it is only returning error.

```suggestion
	if err == nil {
		r.InitialInstances = instances
		volumes, err := GetAvailableVolumes(ctx, svc, clusterTag, clusterTagValue, machineClass, secretData)
		if err == nil {
			r.InitialVolumes = volumes
			r.InitialMachines, err = GetMachines(machineClass, secretData)
			return err
		} else {
			return err
		}
	} 
	return err
```